### ISSUE TITLE
tiling or auto arrange for windows: feature request

### ISSUE THREAD

**Issue by nslieske** [ekzhang/sshx#89]
It would be nice to see an auto arrange of terminals in a systematic format:

1x3
2x2
4x4
etc.

Different aspect ratios
Auto-tiling

I pictured something like i3 windows manager. 

Keyboard short-cuts to add windows. 

Side note: my daughter (15) saw the set-up on 2 * 4k monitors with about 20 terminals open and lots of activity.  She was very impressed and said it just needs windows for tiktok, instagram, and other socials. Just a joke, but obviously very interesting if we can extend xwindow apps.

**Comment by ekzhang** [comment]
Makes sense! Automatic tiling is a good idea, it could live up there in one of the menus. Will need some creative direction in terms of how the tiling algorithm works.

**Comment by nslieske** [comment]
I'm fond of i3:
https://i3wm.org/docs/userguide.html#_window_title_alignment

Tmux, screen, or Zellij all come to mind with their various keyboard short-cuts.

I installed this earlier and found it to be not bad.  Tabbing feature would be very nice and the 14 window limitation gets challenging in sshx.

https://zellij.dev/

Key bindings at the browser level sounds like lots of conflicts with browser & OS.

A quick and dirty feature might be a drop down from the insert + sign that created 2,4,6,8 in a grid to get started.  

![image](https://github.com/user-attachments/assets/11217d8f-1bd9-4c8b-8b91-c222aa1b2c43)

Auto arrange with a key stroke would be awesome.

Let me know how I can help.

**Comment by ekzhang** [comment]
Very cool examples to take some inspiration from. Yeah, let's think about it.

I'd rather not mess with the size of any user's terminal windows since they probably were intentional about resizing them if they did that, so maybe we can figure out some kind of geometry-aware layout algorithm.

**Comment by ekzhang** [comment]
> and the 14 window limitation gets challenging in sshx

Unfortunately this is a browser graphics card limit

**Comment by nslieske** [comment]
> > and the 14 window limitation gets challenging in sshx
> 
> Unfortunately this is a browser graphics card limit

How about tabs as an addition / alternative? If there were tabs, then 14 max is plenty.

**Comment by nslieske** [comment]
If you can lay out the design, I can put some guys to work on it.
